### PR TITLE
[FRAME-3] Add default template for Angular

### PR DESCRIPTION
**Pull request by janvennemann**
**JIRA:** https://jira.appcelerator.org/browse/FRAME-3

**Optional Description:**
This adds the default project template for Angular that will be used with the new `--ng` switch (see https://github.com/appcelerator/appc-cli-titanium/pull/258)

### CODE REVIEW

File: templates/app/angular-default/hooks/post-create.js
Comment (ewanharris) [inline templates/app/angular-default/hooks/post-create.js]: Executing npm (or similar) commands through spawn is problematic, usually ending up with an ENOENT. I think one of the following should suffice (personally I think `shell: true`)

* Pass `shell: true` in the options of the spawn call. Only supported on Node 5.7.0 and up and we do support Node 4, but looking at the code I think it's safe to maybe error if you're using less than Node 8 or 6?
* Use exec instead of spawn (I think). But might run into maxbuffer issues
* Do the same as `shell: true` manually. Something like `spawn('cmd', ['/s', '/c', 'npm', 'i'],{ cwd: path.dirname(packageJsonPath) });`

**Review by ewanharris — COMMENTED** [review]
Comment around spawning npm on Windows

**Review by hansemannn — APPROVED** [review]
Project wide changes are looking fine! Awaiting @ewanharris' approval to proceed. Also being blocked by the [appc-cli PR](https://github.com/appcelerator/appc-cli-titanium/pull/258). @ewanharris Can you review those as well?

**Review by ewanharris — APPROVED** [review]
Creating a project works on Windows, building to Android looks good, Windows (as a build target) has the known issue which I will file a ticket for.

**Pull request merged**
